base_layer/core/src/base_node/comms_interface/inbound_handlers.rs

```diff
@@ -461,7 +471,16 @@ where B: BlockchainBackend + 'static
             source_peer
         );
 
+        // We need to make sure that we can ask for the block from multiple sources as a single source can maliciously
+        // delay the block blocking us from getting to true latest tip
         let block = self.reconcile_block(source_peer.clone(), new_block).await?;
+
+        {
+            let mut lock = self.list_of_reconciling_blocks.lock().await;
+            let list_of_reconciling_blocks = lock.deref_mut();
+            list_of_reconciling_blocks.remove(&block_hash);
```

Comment:
nit: deref_mut is implicitly called
```suggestion
            lock.remove(&block_hash);
```